CMSgov/easi-app
# Easi 4328/access control

# EASI-4328

## Description

- loosen auth to allow for GRB viewers/voters to access system intakes
- remove multi-return where unused (needed for this PR)
- remove some unused functions


## How to test this change

- try to load a system intake you should NOT have access to
- add yourself as a GRB reviewer
- try again to load, it should work
- example flow: 

1. locally, log in with `ABCD`, create system intake, copy system intake `id` and store
2. log in with `FAUI` (`EASI_P_USER` only) and attempt to view that system intake (`http://localhost:3000/governance-task-list/<intake UUID>`)
3. it should NOT load
4. in db, run this query
```sql
INSERT INTO system_intake_grb_reviewers (
  id,
  user_id,
  system_intake_id,
  voting_role,
  grb_role,
  created_by,
  created_at,
  modified_by,
  modified_at
) VALUES (
	gen_random_uuid(),
 	(SELECT id FROM user_account WHERE username = 'FAUI'),
 	'<intake UUID>',
	'NON_VOTING',
 	'CO_CHAIR_CIO',
 	(SELECT id FROM user_account WHERE username = 'ABCD'),
	NOW(),
	(SELECT id FROM user_account WHERE username = 'ABCD'),
	NOW()
);
``` 
5. refresh the page from earlier (`http://localhost:3000/governance-task-list/<intake UUID>`) - it should work!

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
